packages/build/src/plugins_core/blobs_upload/utils.ts

```diff
@@ -0,0 +1,66 @@
+import fsPromises from 'node:fs/promises'
+import path from 'node:path'
+
+import { fdir } from 'fdir'
+
+import { getBlobsDir } from '../../utils/blobs.js'
+
+const METADATA_PREFIX = '$'
+const METADATA_SUFFIX = '.json'
+
+export const anyBlobsToUpload = async function ({ buildDir, publishDir }) {
+  const blobsDir = getBlobsDir({ buildDir, publishDir })
+  const { files } = await new fdir().onlyCounts().crawl(blobsDir).withPromise()
+  return files > 0
+}
+
+/* Given output directory, find all file paths to upload excluding metadata files */
+export async function getKeysToUpload(blobsDir: string): Promise<string[]> {
+  const files = await new fdir()
+    .withRelativePaths() // we want the relative path from the blobsDir
+    .filter((fpath) => !path.basename(fpath).startsWith(METADATA_PREFIX))
+    .crawl(blobsDir)
+    .withPromise()
+
+  // normalize the path separators to all use the forward slash
+  return files.map((f) => f.split(path.sep).join('/'))
+}
+
+/* Read a file and its metadata file from the blobs directory */
```

Comment:
```suggestion
/** Read a file and its metadata file from the blobs directory */
```

A valid JsDoc comment needs to start with two asterisks. Otherwise the intellisense cannot pick it up 😬